languagetool-language-modules/ar/src/main/java/org/languagetool/rules/ar/ArabicTransVerbDirectToIndirectRule.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * LanguageTool, a natural language style checker
+ * Copyright (C) 2022 Sohaib Afifi, Taha Zerrouki
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ar;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedToken;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.language.Arabic;
+import org.languagetool.rules.*;
+import org.languagetool.synthesis.ar.ArabicSynthesizer;
+import org.languagetool.tagging.ar.ArabicTagManager;
+import org.languagetool.tagging.ar.ArabicTagger;
+
+import java.util.*;
+
+import static java.lang.Math.min;
+
+/**
+ * TODO : Sohaib - check the duplicated code
```

Comment:
should this be done before merging the PR?

Reply:
Actually, many codes come from the English part. I was thinking of a way to reduce such duplications.  

Reply:
Yes, please clean up the duplication before we merge the PR.